packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -10,13 +10,36 @@ pub struct PostgresDatabase {
 }
 
 impl PostgresDatabase {
+    /// Crates a new `PostgresDatabase` connection.
+    ///
+    /// # Errors
+    ///
+    /// If creating a [`PgPool`] connection returns an error
+
+    pub async fn new(
+        user: &str,
+        password: &str,
+        host: &str,
+        port: u16,
+        database: &str,
+    ) -> Result<Self, sqlx::Error> {
+        Self::from_url(&format!(
+            "postgres://{user}:{password}@{host}:{port}/{database}"
+        ))
+        .await
+    }
+
     /// Creates a new `PostgresDatabase` object
     ///
     /// # Errors
+    ///
     /// If creating a [`PgPool`] connection returns an error
-    pub async fn new(connect_url: &str) -> Result<Self, sqlx::Error> {
+    pub async fn from_url(connect_url: &str) -> Result<Self, sqlx::Error> {
         Ok(Self {
-            pool: PgPool::connect(connect_url).await.report()?,
+            pool: PgPool::connect(connect_url)
+                .await
+                .report()
+                .attach_printable_lazy(|| format!("Could not connect to {connect_url}"))?,
```

Comment:
I don't think this should block the PR but I think we should continue to think about how we use attachments, we didn't intend them to just be format strings as that's still embedding important information (like the URL) inside of plain-text, which means they're not programmatically accessible. We kind of wanted to be able to do `attach!(url=connect_url)` or something

Reply:
I agree, that we probably shouldn't use `attach_printable` here. However, your proposed tracing-syntax doesn't work because we don't have a key (i.e. `url=`) here. I will make a few small adjustments and then we can decide if this is better.